extensions-contrib/kubernetes-overlord-extensions/src/main/java/org/apache/druid/k8s/overlord/KubernetesTaskRunner.java

```diff
@@ -206,6 +217,31 @@ protected TaskStatus doTask(Task task, boolean run)
     }
   }
 
+  @VisibleForTesting
+  protected void listenOnTaskState(KubernetesPeonLifecycle.State state, String taskId)
```

Comment:
Might as well just call this method `emitTaskStateMetrics` as it is not doing much else right now.

Reply:
Updated.